### ISSUE TITLE
[HOLD for payment 2023-09-21] [HOLD] [$1000] Chat - LHN page displayed instead Legal name page when using deeplink

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23745]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login in the same account in 2 platform (mWeb/Safari and iOS app)
2. Go to Settings -> Profile -> Personal details -> Legal name -> Copy the URL of page  (mWeb/Safari)
3. Paste URL in another program
4. Log out in iOS app
5. Open the URL that was copied in step 2
6. Login with same account

## Expected Result:
 App navigates user to the Legal name page 

## Actual Result:
LHN displayed instead of the Legal name page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.46-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/352083fa-174d-4575-8ed5-9ca71fd65d17

**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal team
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01327557008996edfb</li>
        <li>Upwork Job ID: 1688707539214979072</li>
        <li>Last Price Increase: 2023-08-08</li>
<li>Automatic offers: </li>
<ul>
<li>mollfpr | Reviewer | 26045352</li>
</ul></li>
    </ul>
</details>

**Comment by Talha345** [comment]
@kbecciv I think the task description does not conform with the video atatched. Can you please recheck and clarify?

**Comment by jliexpensify** [comment]
Agree with @Talha345 . @kbecciv please re-test and attach the correct video and re-open this GH.

**Comment by kbecciv** [comment]
Hey @jliexpensify Checking

**Comment by kbecciv** [comment]
@jliexpensify Sorry about confusion, steps and video have been updated. Thank you

**Comment by jliexpensify** [comment]
@kbecciv sorry I am still confused: you mentioned that this is for the IOS app but you've also checked Android. So is this an iOS issue or an Android one?

**Comment by jliexpensify** [comment]
Bumop @kbecciv - is this an iOS issue only or for both platforms? Thanks.

**Comment by jliexpensify** [comment]
Can't reproduce on Pixel 3a (Android) - takes me to the legal page on 1.3.48-0.

@kbecciv please tag me if you can reproduce on iOS and share a video.

**Comment by puneetlath** [comment]
I just tested this on iOS and the deep link does not seem to be working.

**Comment by jliexpensify** [comment]
Thank you Puneet, seems to be an iOS issue only then.

**Comment by jliexpensify** [comment]
Whoops sorry - this is open to proposals. Have updated the title!

**Comment by waterim** [comment]
Hello, Im Artem from Callstack - expert group. Can I help you with this one?

**Comment by kubabutkiewicz** [comment]
Hi, I'm Jakub from Callstack - expert contributor group - and I would like to take a look at this issue.

**Comment by mollfpr** [comment]
@jliexpensify I guess we can HOLD this for the proposal review above; if not 🎀 then we can assign it to @kubabutkiewicz

**Comment by jliexpensify** [comment]
@mollfpr sorry I am a little confused - are you proposing we go with @tienifr ? If so, you'd be assigning an Engineer right?

**Comment by mollfpr** [comment]
@jliexpensify I mean; we can HOLD the issue to prevent more proposals from coming.

> Are you proposing we go with @tienifr

If their proposal is the solution. If not we can assign the issue immediately to the agency.

**Comment by jliexpensify** [comment]
Ok, so I've dropped a HOLD on the title and assigned @kubabutkiewicz (as per the process with Callstack) - let us know how you go.

@tienifr - I don't see any prior proposals from you in this GH, [just this one](https://github.com/Expensify/App/issues/23745#issuecomment-1671286441) which was posted after @kubabutkiewicz expressed interest. If I'm missing something, please feel free to link me where you made the prior proposal.

@mollfpr let me know if am I missing anything!

**Comment by kubabutkiewicz** [comment]
Just FYI the problem occurs on both iOS and Android 

https://github.com/Expensify/App/assets/25230417/18686c65-f090-4860-88e1-ae0f7a42debe

**Comment by mollfpr** [comment]
@tienifr 

> This case is the one which has the issue, this is because at the time we try to navigate to the deep link, the route is not present in the navigation structure of the app, so the deep linking fail.

I feel this might be correct, but it's not working only on the native. Why it's working on another platform?

Although your solution may fix this issue, it's not fixing the root cause. @kubabutkiewicz Can I get your opinion on @tienifr [proposal](https://github.com/Expensify/App/issues/23745#issuecomment-1671286441) too?

**Comment by kubabutkiewicz** [comment]
@mollfpr Sure thing!

So from my observation this is exactly the problem when we are signing out from the app we are setting navigation state to point to "public" screens. 
But the fix from @tienifr seems to me more like a workaround I think it should be handled similar to [this](https://github.com/Expensify/App/blob/main/src/components/DeeplinkWrapper/index.website.js) component.

**Comment by kubabutkiewicz** [comment]
Alternatively what we can do is to force navigation to reset completely state after signing out to behave like we are opening app from completely closed app.

@mollfpr let me know what you think !

**Comment by kubabutkiewicz** [comment]
@mollfpr 
After further investigation I see that is more as a feature not a bug , since none of the deeplinks are working when app is in the background state expect one which is handled by this listener which @tienifr provided.

**Comment by kubabutkiewicz** [comment]
Yeah sorry, that what I was thinking , maybe bad wording from my side 😅

**Comment by jliexpensify** [comment]
@tienifr sorry, specifically what I'm wondering about are this statement:

>I've spent many hours investigating the issue

I appreciate that you may have started to troubleshoot this before @kubabutkiewicz indicated interest, but it's our internal process to assign issues to someone from an agency if they indicate interest on a GH _before_ any Contributors have made a proposal.

If there have been proposals, the C+ is encouraged to review those first, and if none are desirable, then the person from the agency is assigned.

It seems that @mollfpr has reviewed your proposal though, and mentioned that your issue doesn't fix the root cause, so it wasn't accepted - is this correct?

**Comment by jliexpensify** [comment]
Sure, I'm happy for you and @mollfpr to discuss this, along with @kubabutkiewicz's input  - ultimately, I think @mollfpr should make the judgment call here. @kubabutkiewicz was assigned as per the internal processes (this is to stop other Contributors posting proposals/being assigned).

**Comment by kubabutkiewicz** [comment]
Update from me for today. I think I found root cause , I am discussing internally best solution to that.

**Comment by kubabutkiewicz** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Deeplinks are not navigating to correct screens when app is in the background mode
### What is the root cause of that problem?
The root cause of the problem is that Linking listener which is in Expensify.js file is made in the way that is handling only deeplinks to Report screen and to Concierge screen
### What changes do you think we should make in order to solve the problem?
Currently action **openReportFromDeepLink** from ReportUtils is responsible to solve this but its solve this only for some specifics deeplinks. What we can do here is to add there **Navigation.navigate** add pass there path from a deeplink. **Navigation.navigate** is using under the hood linkTo which is using **linkingConfig** to navigate users. Doing that this action will be more generic and will handle all possible deeplinks not only specific.
### What alternative solutions did you explore? (Optional) (edited)

**Comment by kubabutkiewicz** [comment]
@mollfpr 
I will be ooo next week so somebody from Callstack will take over this task , I already created PR with change for next person

**Comment by lukemorawski** [comment]
@mollfpr Hey. It's Lucas from Callstack. Happy to take over

**Comment by mollfpr** [comment]
Thanks @kubabutkiewicz for the notice.

Heyy @lukemorawski, let's tackle this issue 🚀

**Comment by hayata-suenaga** [comment]
### Memo
@lukemorawski from CallStack is going to work on this

**Comment by kubabutkiewicz** [comment]
@hayata-suenaga Yes I am also from Callstack :) 
I was on OOO and @lukemorawski took over this task from me but I am back now so I could take it again :)

**Comment by mollfpr** [comment]
>I appreciate that you may have started to troubleshoot this before @kubabutkiewicz indicated interest, but it's our internal process to assign issues to someone from an agency if they indicate interest on a GH before any Contributors have made a proposal.

>If there have been proposals, the C+ is encouraged to review those first, and if none are desirable, then the person from the agency is assigned.

@kubabutkiewicz Already put their interest before your proposal was submitted. The process should be they get assigned, and the C+ will not review any proposal (_only review proposals that were submitted before the agency came_). 

@tienifr We already gave you a chance, but we only have a little time to get the desired solution. For the compensation, I'm not eligible to give you the answer.

**Comment by jliexpensify** [comment]
I'm not an Engineer so I can't speak to how much of your proposal "inspired" the PR (@hayata-suenaga or @mollfpr can weigh in) - but my instinct, as a Bug Zero member managing payments to follow the processes, is that we wouldn't pay you.

If the external Contributor was struggling and you actually helped them come up with the proposal, then I'd be inclined to split the payment 50/50.

**Comment by hayata-suenaga** [comment]
wow a lot of conversations happening here

@mollfpr if you could, could you summarize what is being discussed and what needs to be decided?

**Comment by kubabutkiewicz** [comment]
> If the external Contributor was struggling and you actually helped them come up with the proposal
> I think @kubabutkiewicz is the one who can answer this.

When I was looking at this issue I wasn't inspired by what @tienifr, I just looked at his proposal when @mollfpr asked me and gave my thoughts. 
The reason which @tienifr suggested as a root cause was actually good but the solution which was suggested was not good for me thats why I said this looks like a workaround.

**Comment by kubabutkiewicz** [comment]
@tienifr 
What I meant is that I did not suggest by your proposal and just did my own. 
So your investigation was good.
That's all. 
I have finger crossed that you will be compensated :D

**Comment by hayata-suenaga** [comment]
@mollfpr did the PR used most of @tienifr's suggestion?

@jliexpensify If this is the case, we can do partial compensation for @tienifr. I remember I read somewhere that there is an instruction on what to do when an internal engineer takes a contributor's proposal and implement it themselves. I think we can apply the same guideline here for the payment amount.

**Comment by mollfpr** [comment]
@hayata-suenaga The idea is the same, but the implementation is quite different.

**Comment by jliexpensify** [comment]
Hi @tienifr - I've spoken privately to @mollfpr and @hayata-suenaga to understand the situation presented here, and here's a summary of our joint decision:

- Firstly, we appreciate that you spent time on this issue, but it’s in the [CONTRIBUTING.md](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#for-external-agencies-that-expensify-partners-with) that as soon as an agency comments, we’ll assign to them. This is the same for any issue out there.

- Similarly, we appreciate that you had a proposal first, but it wasn’t accepted. @mollfpr also confirmed with @kubabutkiewicz that their PR was not inspired by your solution. It has some overlap, but [ultimately](https://github.com/Expensify/App/pull/26073#issuecomment-1696988237) didn’t solve for the root cause.

**- To close off this issue, we feel it's fair to offer a one-time payment of $250 to you - to recognise there is some overlap.**

To be transparent, we all feel like this debate over compensation went on for too long and felt like we adequately explained our reasoning to you: These situations are always tricky to navigate, especially when there is compensation involved. We definitely appreciate you have worked on a proposal, but we're also bound by the processes set out in the .MD and by our wider team. It would be unfair to bend the rules for one contributor and goes against Rule #2 - Don't Ruin It For Everybody.

I will also take this opportunity to remind you that even though a proposal is posted, any issue can be ultimately made Internal or the proposal posted not be considered/used.

I'll go ahead and create an Upworks job and arrange for a payout to you.

**Comment by jliexpensify** [comment]
Bump @mollfpr to complete the checklist!

**Comment by mollfpr** [comment]
> [@mollfpr] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/15788

> [@mollfpr] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/15788/files#r1330344843

> [@mollfpr] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

The regression step should be enough.

>[@mollfpr] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

1. Make sure the app is in the background mode (only for iOS or Android)
2. Make sure you are logged out
3. Open the app from whatever deep link
4. Log in
5. You should be redirected to the screen that you provided in the deep link
6. 👍 or 👎

**Comment by mollfpr** [comment]
Friendly bump @jliexpensify

**Comment by jliexpensify** [comment]
Thanks for the bump - payment summary:

- Contributor: @kubabutkiewicz $1000 (Invoicing)
- Contributor: @tienifr $250 ([special payment, discussed here](https://github.com/Expensify/App/issues/23745#issuecomment-1710879931))
- C+: @mollfpr $1000

[Upworks job](https://www.upwork.com/jobs/~01327557008996edfb)

Am I missing anything?

**Comment by mollfpr** [comment]
@jliexpensify I already have the Upwork contract for this issue.

**Comment by jliexpensify** [comment]
Yep, I'm aware @mollfpr - I generally do a payment summary for every job now, to make sure that I'm paying the correct amounts. Will pay via Upworks now.

**Comment by jliexpensify** [comment]
All paid, job closed
